File: java-dagger2/app/src/main/java/com/jetictors/futures/java_dagger2/di/component/NamedComponent.java
package com.jetictors.futures.java_dagger2.di.component;

import com.jetictors.futures.java_dagger2.act.NamedActivity;
import com.jetictors.futures.java_dagger2.di.modules.NamedModule;

import dagger.Component;

/**
 * @author :   Jetictors
 * @version :   v1.0.1
 * @描述 :    Named注解用法的Component
 * @time :   2018/2/7 15:53
 */

@Component(modules = NamedModule.class)
public interface NamedComponent {
    void inject(NamedActivity activity);
}

File: kotlin-dagger2/app/src/main/java/com/jetictors/futures/kotlin_dagger2/di/component/NamedComponent.kt
package com.jetictors.futures.kotlin_dagger2.di.component

import com.jetictors.futures.kotlin_dagger2.act.NamedActivity
import com.jetictors.futures.kotlin_dagger2.di.module.NamedModule
import dagger.Component

/**
 * @author :   Jetictors
 * @version :   v1.0.1
 * @描述 :    Named注解用法的Component
 * @time :   2018/2/7 15:53
 */

@Component(modules = arrayOf(NamedModule::class))
interface NamedComponent {

    fun inject(activity: NamedActivity)

}
